scipy/linalg/decomp.py

```diff
@@ -371,126 +415,170 @@ def eigh(a, b=None, lower=True, eigvals_only=False, overwrite_a=False,
     True
 
     """
+    # set lower
+    uplo = 'L' if lower else 'U'
+    # Set job for Fortran routines
+    _job = (eigvals_only and 'N') or 'V'
+
+    drv_str = [None, "ev", "evd", "evr", "evx", "gv", "gvd", "gvx"]
+    if driver not in drv_str:
+        raise ValueError('"{}" is unknown. Possible values are "None", "{}".'
+                         ''.format(driver, '", "'.join(drv_str[1:])))
+
     a1 = _asarray_validated(a, check_finite=check_finite)
     if len(a1.shape) != 2 or a1.shape[0] != a1.shape[1]:
-        raise ValueError('expected square matrix')
+        raise ValueError('expected square "a" matrix')
     overwrite_a = overwrite_a or (_datacopied(a1, a))
-    if iscomplexobj(a1):
-        cplx = True
-    else:
-        cplx = False
+    cplx = True if iscomplexobj(a1) else False
+    n = a1.shape[0]
+    drv_args = {'overwrite_a': overwrite_a}
+
     if b is not None:
         b1 = _asarray_validated(b, check_finite=check_finite)
         overwrite_b = overwrite_b or _datacopied(b1, b)
         if len(b1.shape) != 2 or b1.shape[0] != b1.shape[1]:
-            raise ValueError('expected square matrix')
+            raise ValueError('expected square "b" matrix')
 
         if b1.shape != a1.shape:
-            raise ValueError("wrong b dimensions %s, should "
-                             "be %s" % (str(b1.shape), str(a1.shape)))
-        if iscomplexobj(b1):
-            cplx = True
-        else:
-            cplx = cplx or False
-    else:
-        b1 = None
+            raise ValueError("wrong b dimensions {}, should "
+                             "be {}".format(b1.shape, a1.shape))
 
-    # Set job for Fortran routines
-    _job = (eigvals_only and 'N') or 'V'
```

Comment:
In gh-11262 you wrote:
> The eigvalsh sounds like a bypass to eigenvector computation but unfortunately the eigenvectors are always computed and discarded at the function exit.

Is that true? 
`eigvalsh` always called `eigh` with the option `eigvals_only=True`, and in that case this line previously set `_job` to `'N'` correctly. `_job` was passed into the `jobv` argument the driver, for which:
```
'N':  Compute eigenvalues only;
```

What am I missing?

Reply:
maybe I read too quickly I'll check in the evening once again.

Reply:
What was the conclusion?

Reply:
Misread :)

Reply:
I was wrong :)

Reply:
The wrapper however returns the zeros array even when the eigenvectors are not requested. So I didnot point to the right place for this. In other words, even if the job was 'N' we were still getting a 1000x1000 array of zeros for eigenvectors in the previous wrappers which basically adds a huge overhead.

Reply:
Ah